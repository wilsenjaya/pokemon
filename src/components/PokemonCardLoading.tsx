/* eslint-disable react/require-default-props */
import {
  Box,
  SkeletonCircle,
  Skeleton,
} from '@chakra-ui/react';

type Props = {
  withoutCircle?: boolean;
};

const PokemonCardLoading = ({ withoutCircle }: Props) => (
  <Box
    p={4}
    boxShadow="xl"
    textAlign="center"
    borderWidth="1px"
    borderRadius="2xl"
  >
    {!withoutCircle && <SkeletonCircle size="130px" m="auto" />}
    <Skeleton height={5} maxWidth={200} mx="auto" mt={4} />
    <Skeleton height={5} maxWidth={200} mx="auto" mt={4} />
    <Skeleton height={5} maxWidth={200} mx="auto" mt={4} />
  </Box>
);

export default PokemonCardLoading;
