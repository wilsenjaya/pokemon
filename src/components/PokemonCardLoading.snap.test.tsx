import renderer from 'react-test-renderer';
import PokemonCardLoading from './PokemonCardLoading';

describe('#PokemonCardLoading Component', () => {
  it('should renders correctly with default props', () => {
    const tree = renderer.create(<PokemonCardLoading />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should renders correctly with withoutCircle props', () => {
    const tree = renderer.create(<PokemonCardLoading withoutCircle />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
