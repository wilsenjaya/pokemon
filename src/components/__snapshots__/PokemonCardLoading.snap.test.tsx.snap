// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#PokemonCardLoading Component should renders correctly with default props 1`] = `
<div
  className="css-x1qbfg"
>
  <div
    className="chakra-skeleton css-16jkguf"
  />
  <div
    className="chakra-skeleton css-fw49k7"
  />
  <div
    className="chakra-skeleton css-fw49k7"
  />
  <div
    className="chakra-skeleton css-fw49k7"
  />
</div>
`;

exports[`#PokemonCardLoading Component should renders correctly with withoutCircle props 1`] = `
<div
  className="css-x1qbfg"
>
  <div
    className="chakra-skeleton css-fw49k7"
  />
  <div
    className="chakra-skeleton css-fw49k7"
  />
  <div
    className="chakra-skeleton css-fw49k7"
  />
</div>
`;
